src/Microsoft.Azure.WebJobs/ParameterBindingData.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// A reference type for supporting SDK-type bindings in out-of-process Function workers
+    /// This type has built-in serialization using System.Text.Json.
+    /// </summary>
+    [JsonConverter(typeof(ParameterBindingData))]
+    public class ParameterBindingData
+    {
+        /// <summary> Initializes a new instance of the <see cref="ParameterBindingData"/> class. </summary>
+        /// <param name="source"> Identifies the extension this event is coming from </param>
+        /// <param name="jsonSerializableData"> An object containing any required information to hydrate an SDK-type object in the out-of-process worker </param>
+        /// <param name="dataSerializationType"> The type to use when serializing the data.
+        /// If not specified, <see cref="object.GetType()"/> will be used on <paramref name="jsonSerializableData"/>.</param>
+        public ParameterBindingData(string source, object jsonSerializableData, Type dataSerializationType = default)
+        {
+            if (jsonSerializableData is BinaryData)
+            {
+                throw new InvalidOperationException("This constructor does not support BinaryData. Use the constructor that takes a BinaryData instance.");
```

Comment:
since it is a public type, consider adding the exceptions this method/constructor throws in the method summary(IDEs usually shows that in the tooltip). [Example ](https://github.com/Azure/azure-functions-dotnet-worker/blob/main/extensions/Worker.Extensions.Abstractions/src/BindingPropertyNameAttribute.cs#L18)